controller/tests/array_action/ds8k/test_array_mediator_ds8k.py

```diff
@@ -387,3 +402,74 @@ def test_get_host_by_identifiers_not_found(self):
             self.array.get_host_by_host_identifiers(
                 Initiators('', ["new_wwpn", "another_wwpn"])
             )
+
+    def test_get_snapshot_not_exist_return_none(self):
+        self.client_mock.get_snapshot.side_effect = [ClientError("400", "BE7A002D")]
+
+        snapshot = self.array.get_snapshot("fake_name")
+
+        self.assertIsNone(snapshot)
+
+    def test_get_snapshot_has_no_fcmap_raise_error(self):
+        self.client_mock.get_pools.return_value = [Munch({'id': 'P0'})]
+        self.client_mock.get_volumes_by_pool.return_value = [self.volume_response]
+
+        with self.assertRaises(array_errors.SnapshotNameBelongsToVolumeError):
+            self.array.get_snapshot("test_name")
+
+    def _get_mapped_target_vol(self):
+        volume = self.volume_response
+        volume.flashcopy = [self.flashcopy_response]
+        return volume
+
+    def test_get_snapshot_get_fcmap_not_exist_raise_error(self):
+        target_vol = self._get_mapped_target_vol()
+        self.client_mock.get_pools.return_value = [Munch({'id': 'P0'})]
+        self.client_mock.get_volumes_by_pool.return_value = [target_vol]
+        self.client_mock.get_flashcopies.side_effect = NotFound("404", message="BE7A0001")
+
+        with self.assertRaises(NotFound):
+            self.array.get_snapshot("test_name")
+
+    def test_get_snapshot_success(self):
+        target_vol = self._get_mapped_target_vol()
+        self.client_mock.get_pools.return_value = [Munch({'id': 'P0'})]
+        self.client_mock.get_volumes_by_pool.return_value = [target_vol]
+        self.client_mock.get_flashcopies.return_value = self.flashcopy_response
+
+        self.array.get_snapshot("test_name")
```

Comment:
we should probably add assert here on return value (also missing in svc).
please add ticket number for after GA

Reply:
done

Reply:
I couldn't find any ticket number.
could you point me to the relevant line, so we could resolve this?